gef.py

```diff
@@ -4635,6 +4653,22 @@ def has_setting(self, name):
         key = self.__get_setting_name(name)
         return key in __config__
 
+    def update_setting(self, name, value, description=None):
+        """
+        Function to allow for internal commands to update the value
+        of a setting without destroying the description.
+        """
+        # make sure settings are always associated to the root command (which derives from GenericCommand)
+        if "GenericCommand" not in [x.__name__ for x in self.__class__.__bases__]:
```

Comment:
Does `issubclass` not work?

Reply:
take a look at `add_setting` function right below mine, I literally just copied how that code did it, so I'm not really sure if you want this modified without modifying that one.

Reply:
@hugsy do you remember why it's this way?